### PR TITLE
Added transpose lowering

### PR DESCRIPTION
**Pull request by gpetters94**
This is the transpose.int lowering. There also seems to be one using Dimnames, but I'm not sure if that's in the scope of what I'm doing. Please let me know if I'm missing anything - this is my first PR to an LLVM project.

### CODE REVIEW

File: lib/Conversion/TorchToLinalg/TorchToLinalg.cpp [hunk 1]
```diff
@@ -1275,6 +1275,89 @@ class ConvertAtenUnsqueezeOp : public OpConversionPattern<AtenUnsqueezeOp> {
...
+    if (dim1 < 0 || dim1 >= inputRank)
+      return rewriter.notifyMatchFailure(op, "dim1 out of range");
+
+    auto loc = op.getLoc();
+
+    llvm::SmallVector<Value> outputDims;
+    for (auto i = 0; i < inputRank; i++) {
+      outputDims.push_back(getDimOp(rewriter, loc, adaptor.self(), i));
+    }
+    auto tmp = outputDims[dim0];
```
Comment (cathyzhyi) [inline lib/Conversion/TorchToLinalg/TorchToLinalg.cpp:1322]: nit: can probably use std::swap to simplify the code a little bit.

File: lib/Conversion/TorchToLinalg/TorchToLinalg.cpp [hunk 2]
```diff
@@ -1275,6 +1275,89 @@ class ConvertAtenUnsqueezeOp : public OpConversionPattern<AtenUnsqueezeOp> {
...
+    outputDims[dim1] = tmp;
+
+    Value outVector =
+        rewriter.create<linalg::InitTensorOp>(loc, outputDims, elementType);
+    SmallVector<AffineExpr> idExprs;
+    SmallVector<AffineExpr> swapExprs;
+    for (auto i = 0; i < inputRank; i++) {
+      idExprs.push_back(getAffineDimExpr(i, rewriter.getContext()));
+    }
+    for (auto i = 0; i < inputRank; i++) {
```
Comment (cathyzhyi) [inline lib/Conversion/TorchToLinalg/TorchToLinalg.cpp:1333]: nit: might want to remove the braces for single line if blocks to make the code look cleaner here.  https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements.

**Review by cathyzhyi — APPROVED** [review]
Thanks for the change! I added some suggestions for formatting. Otherwise LGTM.

**Comment by cathyzhyi** [comment]
> There also seems to be one using Dimnames

I think don't need to worry about other variances yet if they are not in the models we are targeting right now. We are implementing on an as-needed basis.

File: lib/Conversion/TorchToLinalg/TorchToLinalg.cpp [hunk 3]
```diff
@@ -1275,6 +1275,89 @@ class ConvertAtenUnsqueezeOp : public OpConversionPattern<AtenUnsqueezeOp> {
 };
 } // namespace
 
+#include <iostream>
```
Comment (ramiro050) [inline lib/Conversion/TorchToLinalg/TorchToLinalg.cpp:1278]: Don't forget to remove this `include`!

Comment (silvasean) [inline lib/Conversion/TorchToLinalg/TorchToLinalg.cpp:1278]: drive-by comment: remove <iostream>
https://llvm.org/docs/CodingStandards.html#include-iostream-is-forbidden

(note: when debugging, you can use `llvm::errs()` instead of `std::cerr()` and `llvm::outs()` instead of std::cout, so you never really need iostream anyway)

**Comment by silvasean** [comment]
Now that https://github.com/llvm/mlir-npcomp/pull/307 landed, `frontends/pytorch/e2e_testing/torchscript/basic.py` has been moved to `e2e_testing/torchscript/basic.py`. Not sure if the CI will even work with this change because of those movements (sorry for the earthmoving!)

**Comment by gpetters94** [comment]
I've cleaned the code up, thanks for the feedback!

File: lib/Conversion/TorchToLinalg/TorchToLinalg.cpp [hunk 4]
```diff
@@ -1275,6 +1275,84 @@ class ConvertAtenUnsqueezeOp : public OpConversionPattern<AtenUnsqueezeOp> {
...
+
+    Value outVector =
+        rewriter.create<linalg::InitTensorOp>(loc, outputDims, elementType);
+    SmallVector<AffineExpr> idExprs;
+    SmallVector<AffineExpr> swapExprs;
+    for (auto i = 0; i < inputRank; i++) {
+      idExprs.push_back(getAffineDimExpr(i, rewriter.getContext()));
+    }
+    for (auto i = 0; i < inputRank; i++) {
+      if (i == dim0) {
```
Comment (cathyzhyi) [inline lib/Conversion/TorchToLinalg/TorchToLinalg.cpp:1329]: nit: those braces can be removed as well

**Comment by cathyzhyi** [comment]
We prefer to squash the commits (including the ones to address comments) into one :) Also, can you rebase to the latest head? Thanks!

**Comment by gpetters94** [comment]
Should I open a new PR after I've squashed?

**Comment by gpetters94** [comment]
Okay, it's done.
